monkey/tests/unit_tests/infection_monkey/credential_collectors/test_mimikatz_collector.py

```diff
@@ -110,3 +117,19 @@ def test_pypykatz_result_parsing_no_secrets(monkeypatch):
     collected_credentials = collect_credentials()
     assert len(collected_credentials) == 1
     assert collected_credentials == expected_credentials
+
+
+def test_mimikatz_credentials_stolen_event_published(monkeypatch):
+    mock_event_queue = MagicMock(spec=IEventQueue)
+    patch_pypykatz([], monkeypatch)
```

Comment:
Any reason not to put some credentials in here?

Reply:
Just that it won't affect what we're trying to test here.

Reply:
This test may be testing too much. It's testing whether publish was correct (called once with a `CredentialsStolenEvent`), whether the tags are correctly populated, and whether the credentials are correctly populated.

Maybe it should be 3 tests? The one that tests credentials could have credentials in it.

Reply:
Is that overkill? It makes sense to test whether the event is published or not from `collect_credentials()`, but whether events are published with the correct tags seems to be something that should be tested in the event queue tests (which it is) and having a test for whether `CredentialsStolenEvent` is correctly populating its fields is basically testing whether a dataclass is working as expected.

Reply:
Think of the published events as the output of `collect_credentials()`. Instead of returning a value, it pushes events into the publisher.

If it returned events, what would you test?